src/routes/transactions/transactions-history.imitation-transactions.controller.spec.ts

```diff
@@ -100,1393 +105,2586 @@ describe('Transactions History Controller (Unit) - Imitation Transactions', () =
     await app.close();
   });
 
-  function getImitationAddress(address: `0x${string}`): `0x${string}` {
-    // + 2 is to account for the '0x' prefix
-    const prefix = address.slice(0, prefixLength + 2);
-    const suffix = address.slice(-suffixLength);
-    const imitator = `${prefix}${faker.finance.ethereumAddress().slice(prefixLength + 2, -suffixLength)}${suffix}`;
-    return getAddress(imitator);
-  }
-  const chain = chainBuilder().build();
-  const safe = safeBuilder().build();
+  describe('Event spoofing', () => {
+    function parseUnits(value: bigint, decimals: number): bigint {
```

Comment:
Question: Do we need to use `parseUnits` anywhere else in the code? If that’s the case, I’d suggest creating a helper function for it and importing it in the tests.

Reply:
No, we use [`parseUnits` from viem](https://viem.sh/docs/utilities/parseUnits.html). This was an edge case where we needed a numerical implementation.